utopia/predictor.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2015. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+
+from typechecks import require_integer
+import pandas as pd
+
+class Predictor(object):
+    def __init__(
+            self,
+            mhc_model_type,
+            epitope_lengths=[8, 9, 10, 11],
+            padding_around_mutation=None):
+        """
+        Parameters
+        ----------
+        mhc_model_type : type
+            Type of MHC Class I binding prediction model whose
+            initializer must take a list of HLA allele names called 'alleles'
+            and a list of integers called 'epitope_lengths'
+
+        epitope_lengths : list of int
+
+        padding_around_mutation : int, optional
+            Number of amino acids on each side of the mutation to include
+            in candidate epitopes. If not given, then padding_around_mutation
+            is set to one less than the maximum epitope length.
+        """
+        self.mhc_model_type = mhc_model_type
+        self.epitope_lengths = epitope_lengths
+        if min(epitope_lengths) < 1:
+            raise ValueError(
+                "Epitope lengths must be positive integers, got: %s" % (
+                    min(epitope_lengths),))
+        if padding_around_mutation is None:
+            padding_around_mutation = max(epitope_lengths) - 1
+
+        require_integer(padding_around_mutation, "padding around mutation")
```

Comment:
I thought you were moving away from `require` statements? This one feels funky because it's so deep in the method. (If we had a require, I think I'd prefer `require_integer_or_none` or `require_integer(none_okay=True` or something?